rust/app/src/parser/operations/endorsement.rs

```diff
@@ -160,7 +160,12 @@ impl<'b> EndorsementWithSlot<'b> {
         }
 
         let (rem2, endorsement) = Endorsement::from_bytes(rem)?;
-        let length = (length as usize) - 32 - 1 - (rem.len() - rem2.len());
+        let length = (length as usize).checked_sub(32 + 1);
+        let rem_len = rem.len().checked_sub(rem2.len());
+        let length = length
+            .zip(rem_len)
+            .and_then(|(len, rem)| len.checked_sub(rem))
+            .ok_or(ParserError::parser_value_out_of_range)?;
```

Comment:
so we can reduce this with simple `let length = length.checked_sub(rem).ok_or(PaserError::parser_value_out_of_range)?;`